### ISSUE TITLE
Rule Proposal:  `jsx-max-props-per-component`

### ISSUE THREAD

**Issue by makotot** [jsx-eslint/eslint-plugin-react#3716]
Hi 👋  How about limiting the maximum number of props in a component to avoid [Apropcalypse](https://speakerdeck.com/jenncreighton/flexible-architecture-for-react-components?slide=10)?
Similar to `jsx-max-props-per-line`, but per-component rather than per-line and excludes native html elements.

Example of incorrect code when the maximum number is set to 5:
```tsx
<Button
  a='a'
  b='b'
  c='c'
  d='d'
  e='e'
  f='f'
/>
```

Example of correct code when the maximum number is set to 5:
```tsx
<Button
  a='a'
/>
```

**Comment by ljharb** [comment]
what is the benefit? that talk is from 2018 and i haven't heard of any similar complaints since.

**Comment by makotot** [comment]
In terms of frequency of looking, it does not seem to be a useful rule in general. Thank you for your quick response. Understood. I'll make it personal if necessary.
